arrows/vxl/pixel_feature_extractor.cxx

```diff
@@ -0,0 +1,301 @@
+                     d->enable_high_pass_box,
+                     "Enable high_pass_filter filter." );
+  config->set_value( "enable_high_pass_bidir",
+                     d->enable_high_pass_bidir,
+                     "Enable high_pass_filter filter." );
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+pixel_feature_extractor
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Start with our generated vital::config_block to ensure that assumed values
+  // are present. An alternative would be to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  d->enable_color = config->get_value< bool >( "enable_color" );
+  d->enable_gray = config->get_value< bool >( "enable_gray" );
+  d->enable_aligned_edge = config->get_value< bool >( "enable_aligned_edge" );
+  d->enable_average = config->get_value< bool >( "enable_average" );
+  d->enable_color_commonality = config->get_value< bool >(
+    "enable_color_commonality" );
+  d->enable_high_pass_box =
+    config->get_value< bool >( "enable_high_pass_box" );
+  d->enable_high_pass_bidir =
+    config->get_value< bool >( "enable_high_pass_bidir" );
+
+  // Configure the individual filter algorithms
+  d->aligned_edge_detection_filter.set_configuration(
+    config->subblock_view( "aligned_edge" ) );
+  d->average_frames_filter.set_configuration(
+     config->subblock_view( "average" ) );
+  d->color_commonality_filter.set_configuration(
+    config->subblock_view( "color_commonality" ) );
+  d->high_pass_box_filter.set_configuration(
+    config->subblock_view( "high_pass_box" ) );
+  d->high_pass_bidir_filter.set_configuration(
+    config->subblock_view( "high_pass_bidir" ) );
+  write_config_file( config->subblock_view( "high_pass_box" ), "box.conf" );
+}
+
+// ----------------------------------------------------------------------------
+bool
+pixel_feature_extractor
+::check_configuration( vital::config_block_sptr config ) const
+{
+  auto enable_color = config->get_value< bool >( "enable_color" );
+  auto enable_gray = config->get_value< bool >( "enable_gray" );
+  auto enable_aligned_edge =
+    config->get_value< bool >( "enable_aligned_edge" );
+  auto enable_average = config->get_value< bool >( "enable_average" );
+  auto enable_color_commonality = config->get_value< bool >(
+    "enable_color_commonality" );
+  auto enable_high_pass_box =
+    config->get_value< bool >( "enable_high_pass_box" );
+  auto enable_high_pass_bidir =
+    config->get_value< bool >( "enable_high_pass_bidir" );
+
+  if( !( enable_color || enable_gray || enable_aligned_edge ||
+         enable_average || enable_color_commonality || enable_high_pass_box ||
+         enable_high_pass_bidir) )
+  {
+    LOG_ERROR( logger(), "At least one filter must be enabled" );
+    return false;
+  }
+  return true;
```

Comment:
@russelldj I'm going to merge this as is to keep things moving, but the other thing I just noticed that was missing is that you should also call `check_configuration` on all of the sub algorithm that are enabled and return the `&&` of those.  You can do this on a new branch.